src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -524,18 +577,14 @@ export class DashboardAppController {
         },
         queryFilter.getGlobalFilters()
       );
-      // Making this method sync broke the updates.
-      // Temporary fix, until we fix the complex state in this file.
-      setTimeout(() => {
-        queryFilter.setFilters(queryFilter.getGlobalFilters());
-      }, 0);
+      queryFilter.setFilters(queryFilter.getGlobalFilters());
+      updateNavBar();
     };
 
     const updateStateFromSavedQuery = (savedQuery: SavedQuery) => {
       const savedQueryFilters = savedQuery.attributes.filters || [];
       const globalFilters = queryFilter.getGlobalFilters();
-      const allFilters = [...globalFilters, ...savedQueryFilters];
-
+      const allFilters = [...globalFilters, ...savedQueryFilters, ...queryFilter.getAppFilters()];
```

Comment:
What is this change for? It seems like when applying a saved query the current app filters should get cleared out (and this is also what still happens it seems).

Reply:
This was to fix this isssue: https://github.com/elastic/kibana/pull/55443#pullrequestreview-350679801

When there is a saved query, and a filter is applied afterwards, `updateStateFromSavedQuery` is triggered after the filters are applied, so savedQuery filters override the newly applied filters.